### PR TITLE
[Feature] Add missing properties in forum & thread channels

### LINKED ISSUE #2475: [Bug]: `IForumChannel` is not an `INestedChannel` when it can be nested into a category
### Check The Docs

- [X] I double checked the docs and couldn't find any useful information.

### Verify Issue Source

- [X] I verified the issue was caused by Discord.Net.

### Check your intents

- [X] I double checked that I have the required intents.

### Description

Forum channels can be valid nested channels too, but the `IForumChannel` interface does not implement `INestedChannel`. This poses problems when dealing with category management.

### Version

3.8.2

### Working Version

_No response_

### Logs

No logs

### Sample

_No response_

### Packages

`Discord.Net.Core`

### LINKED ISSUE #2502: [Bug]: Calling ChannelExtensions.GetChannelType on SocketForumChannel returns null
### Check The Docs

- [X] I double checked the docs and couldn't find any useful information.

### Verify Issue Source

- [X] I verified the issue was caused by Discord.Net.

### Check your intents

- [X] I double checked that I have the required intents.

### Description

Everything is in the title.

Calling `ChannelExtensions.GetChannelType` on `SocketForumChannel` returns `null` instead of `ChannelType.Forum`.

While I'm well aware that this method can return null because of the type annotation, I find strange that it doesn't handle the case of a `SocketForumChannel`.

### Version

3.8.1

### Working Version

_No response_

### Logs

Stack trace from the sample code:

```
System.InvalidOperationException: Nullable object must have a value.
   at System.Nullable`1.get_Value()
   at bot.Modules.SynchronizationModule.ReadyHandler() in /home/l4p1n/Code/CLI/discord-bot/bot/Modules/SynchronizationModule.cs:line 38
```

The channel type the code crashes on is `SocketForumChannel`. Its contents is irrelevant for this report.

### Sample

```cs
// Note: The prerequisite code to initialize a client has been cut out

// Replace zero with a real, existing forum channel ID.
ulong forumChannelId = 0;
// _client is a DiscordSocketClient
var forumChannel = _client.GetChannel(forumChannelId) as SocketForumChannel;
Console.WriteLine(
    $"Channel type number of channel ID {forumChannel.Name} " +
    $"is {(int)forumChannel.GetChannelType()}"
);

// The snippet throws an exception with a stack trace similar to the
// one provided in the "Logs" section
```

### Packages

Not relevant. I can fill that in if requested.

### LINKED ISSUE #2503: [Bug]: Unable to create forum post if tags are mandatory
### Check The Docs

- [X] I double checked the docs and couldn't find any useful information.

### Verify Issue Source

- [X] I verified the issue was caused by Discord.Net.

### Check your intents

- [X] I double checked that I have the required intents.

### Description

When trying to create a post in a forum channel where tags are mandatory, it fails with the following error code:
"The server responded with error 40067: A tag is required to create a forum post in this channel"
screenshot

### Version

3.8.1

### Working Version

-

### Logs

-

### Sample

```cs
//get guild
var guild = client.GetGuild(1034041984710627328);
//get forum channel
var forum = guild.GetForumChannel(1034381738442498078);
//try to create a forum post while tags are mandatory
await forum.CreatePostAsync("test");
```

### Packages

None

### PR DESCRIPTION
**Pull request by Misha-133**
This PR adds support for missing channel properties and improves channel modification

Also it fixes a bug in `DiscordRestClient`

### Changes
- [x] new properties for editing channels:
  - `ThreadChannelProperties`
  - `ForumChannelProperties`
- [x] move `Locked` and `Archived` props from `TextChannelProperties` into `ThreadChannelProperties` **Might be breaking**
- [x] add `AppliedTags` property to:
  -  `IThreadChannel`
  - `RestThreadChannel`
  - `SocketThreadChannel`
  - `ThreadChannelProperties`
- [x] add `DefaultSlowModeInterval` and `ThreadCreationInterval` properties
  - to forum channels
  - to `ForumChannelProperties`
- [x] add support for `default_reaction_emoji`
- [x] add `ChannelFlags`
- [x] add `ForumTagBuilder`
- [x] add comparison operators on forum tags
- [x] add `CreateForumChannelAsync` methods to guilds
- [x] add `ForumChannels` property to `SocketGuild`
- [x] add support for `default_sort_order`\
- [x] support creating post with applied tags (also fixes #2503) **Might be breaking**

### Fixes
- [x] Fix `DiscordRestClient.GetChannelAsync` returning `null` if getting forum channel
- [x] Fix #2475
- [x] Fix misleading comment on `ApplicationCommandOptionType.Integer` 
- [x] Fix #2502

### CODE REVIEW

**Review by csmir — APPROVED** [review]

**Comment by JavidPack** [comment]
Looking forward to this functionality!

**Comment by budgetdevv** [comment]
Same, actually running a fork of this PR...hopefully they merge soon

**Review by MrCakeSlayer — APPROVED** [review]

File: src/Discord.Net.WebSocket/Entities/Channels/SocketForumChannel.cs [hunk 1]
```diff
@@ -46,46 +70,68 @@ internal override void Update(ClientState state, Model model)
...
+
+            if (model.DefaultReactionEmoji.IsSpecified && model.DefaultReactionEmoji.Value is not null)
+            {
+                if (model.DefaultReactionEmoji.Value.EmojiId.HasValue && model.DefaultReactionEmoji.Value.EmojiId.Value != 0)
+                    DefaultReactionEmoji = new Emote(model.DefaultReactionEmoji.Value.EmojiId.GetValueOrDefault(), null, false);
+                else if (model.DefaultReactionEmoji.Value.EmojiName.IsSpecified)
+                    DefaultReactionEmoji = new Emoji(model.DefaultReactionEmoji.Value.EmojiName.Value);
+                else
+                    DefaultReactionEmoji = null;
+            }
```
Comment (itsHeroYT) [inline src/Discord.Net.WebSocket/Entities/Channels/SocketForumChannel.cs:93]: @Misha-133 I think you forgot to assign CategoryId in here

Comment (Misha-133) [reply src/Discord.Net.WebSocket/Entities/Channels/SocketForumChannel.cs:93]: Oh, huh, i actually did
Thanks for reminding to add it :)

**Comment by budgetdevv** [comment]
Rejoice!

**Pull request merged**